src/components/heartrate/Ppg.cpp

```diff
@@ -5,45 +5,42 @@
 */
 
 #include "components/heartrate/Ppg.h"
-#include <vector>
 #include <nrf_log.h>
+#include <vector>
 using namespace Pinetime::Controllers;
 
 /** Original implementation from wasp-os : https://github.com/daniel-thompson/wasp-os/blob/master/wasp/ppg.py */
-namespace {
-  int Compare(int8_t* d1, int8_t* d2, size_t count) {
-    int e = 0;
-    for (size_t i = 0; i < count; i++) {
-      auto d = d1[i] - d2[i];
-      e += d * d;
-    }
-    return e;
-  }
-
-  int CompareShift(int8_t* d, int shift, size_t count) {
-    return Compare(d + shift, d, count - shift);
-  }
-
-  int Trough(int8_t* d, size_t size, uint8_t mn, uint8_t mx) {
-    auto z2 = CompareShift(d, mn - 2, size);
-    auto z1 = CompareShift(d, mn - 1, size);
-    for (int i = mn; i < mx + 1; i++) {
-      auto z = CompareShift(d, i, size);
-      if (z2 > z1 && z1 < z)
-        return i;
-      z2 = z1;
-      z1 = z;
-    }
-    return -1;
-  }
-}
-
 Ppg::Ppg()
   : hpf {0.87033078, -1.74066156, 0.87033078, -1.72377617, 0.75754694},
     agc {20, 0.971, 2},
     lpf {0.11595249, 0.23190498, 0.11595249, -0.72168143, 0.18549138} {
 }
 
+int Ppg::Compare(int8_t* d1, int shift, size_t count) {
```

Comment:
Why was this function changed, apart from adding `getRingIndex()`?

Reply:
in the original implementation the pointer was moved forward and then accessed as a 0-based array. That wouldn't work in the ring buffer, because the actual index has to be known to modulo it.